### ISSUE TITLE
Forbid empty comments

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1711]
# Rule request

## Thesis

We should ban comments like this one:

```python
# 
my_var = 1
```

But, empty lines in the middle are fine:

```python
# First line
# 
# Samples:
# One
# Two
my_var = 1
```

We ban only empty lines in the beginning and in the end.

## Reasoning

Why not?

**Comment by Elton-Lin** [comment]
Hello there,
@huangyunzen and I would like to work on this new rule. We have just started with the setup in the contribution page. We will report back when we get things ready and develop an initial plan. Thanks!

**Comment by sobolevn** [comment]
Thanks a lot!

**Comment by Elton-Lin** [comment]
We'd like to get some clarifications for the alerting style for consecutive empty comments. We think that it might make sense to only alert on the first empty line in this case, examples shown below:
```
#           <--- alert
#
my_var = 5
#           <--- alert
#
```

```
#           <--- alert
#
# A cat
#           <--- alert
#
```

What are you thoughts?

**Comment by sobolevn** [comment]
Yes, the second one is fine.

**Comment by sobolevn** [comment]
In other words: we ban only empty lines with any extra associated content.

**Comment by Elton-Lin** [comment]
> Yes, the second one is fine.

Could you elaborate on this? Did you mean we shouldn't alert on my second example because there is actual content within the block of comments? I thought empty comments are only allowed if they are wrapped with actual content around like the example you gave previously. Thanks

**Comment by sobolevn** [comment]
@Elton-Lin updated, sorry. I have to read more carefully 🙂

**Comment by Elton-Lin** [comment]
Thanks for the update. Just to be extra sure, if we only alert on the first line for my second example, does it imply the following case is valid? (I'm guessing no) or we would still alert line 2. (essentially we focus on one block of comments at a time, so there will only be one alert for a block of comments, if needed)

```
# A cat
#           <--- alert or not
#
```

**Comment by sobolevn** [comment]
Yes, this should also alert.

**Comment by Elton-Lin** [comment]
For writing unit tests under tests/test_visitors/test_tokenize/test_comments/, we are wondering if we need to assert the line number of the alert besides confirming the correct violation is created. If so, what would be the best way to do that ? (extracting the line number of the violation and assert with the expected one)

**Comment by sobolevn** [comment]
No, we don't usually assert the line number. We just assert that violation is there.

**Comment by Elton-Lin** [comment]
Thanks for the answer. We are pretty close to having a complete solution, but we are currently failing a style rule about code complexity:
![Screen Shot 2020-11-24 at 12 13 36 PM](https://user-images.githubusercontent.com/33791310/100128675-90681880-2e4e-11eb-9ecd-ac71cfb563da.png)
We simplified the code and tried adding helper function, but there is also an upper limit to the number of functions in a class. What are your suggestions? (Here is our forked [repo](https://github.com/Elton-Lin/wemake-python-styleguide))
